src/azure-cli/azure/cli/command_modules/resource/custom.py

```diff
@@ -72,6 +72,38 @@ def _build_resource_id(**kwargs):
         return None
 
 
+def _try_parse_key_value_object(template_param_defs, parameters, value):
+    # support situation where empty JSON "{}" is provided
+    if value == '{}' and not parameters:
+        return True
+
+    try:
+        key, value = value.split('=', 1)
+    except ValueError:
+        return False
+
+    param = template_param_defs.get(key, None)
+    if param is None:
+        raise CLIError("unrecognized template parameter '{}'. Allowed parameters: {}".format(key, ', '.join(sorted(template_param_defs.keys()))))
```

Comment:
```suggestion
        raise InvalidArgumentValueError("unrecognized template parameter '{}'. Allowed parameters: {}".format(key, ', '.join(sorted(template_param_defs.keys()))))
```
How about using the specific error type `InvalidArgumentValueError` instead of `CLIError`?